File: java/challenges/mergeTwoLinkedLists.java
final class MergeTwoLinkedLists {

 // Singly-linked lists are already defined with this interface:
// class ListNode<T> {
//   ListNode(T x) {
//     value = x;
//   }
//   T value;
//   ListNode<T> next;
// }
//
    ListNode<Integer> append(ListNode<Integer> ln, int data) {
        if(ln == null) {
            ln = new ListNode(data);
            return ln;
        }
        ListNode<Integer> n = ln;
        while(n.next != null) { n = n.next; }
        n.next = new ListNode(data);
        return ln;
    }

    ListNode<Integer> mergeTwoLinkedLists(ListNode<Integer> l1, ListNode<Integer> l2) {
        ArrayList<Integer> list = new ArrayList();
        Integer[] nums = null;
        ListNode<Integer> out = null;
        while(l1 != null) { list.add(l1.value); l1 = l1.next; }
        while(l2 != null) { list.add(l2.value); l2 = l2.next; }
        nums = list.toArray(new Integer[list.size()]);
        Arrays.sort(nums);
        for(int i : nums) { out = append(out, i); }
        return out;
    }
}

File: InterviewPratice/Javascript/mergeTwoLinkedLists.js
// Singly-linked lists are already defined with this interface:
// function ListNode(x) {
//   this.value = x;
//   this.next = null;
// }
//
function mergeTwoLinkedLists(l1, l2) {
    if(l1 == null) return l2;
    
    if(l2 == null) return l1;
    
    if(l1.value < l2.value) {
        l1.next = mergeTwoLinkedLists(l1.next, l2);
        return l1;
    }
    
    l2.next = mergeTwoLinkedLists(l1, l2.next);
    return l2;
}